---
title: "R_Assignment"
date: 3/14/2025
output:
  html_document: default
  pdf_document: default
author: Zheyuan Zhang
---
### 1. Check the enviomental frame
```{r}
library("tidyverse")
```
### 2. read files
```{r}
raw_genotypes <- read_tsv("fang_et_al_genotypes.txt")
raw_snp_position <- read_tsv("snp_position.txt")

```
### 3. delete irrelevant columns
```{r}
genotypes <- raw_genotypes %>% select(-JG_OTU)
snp_position <- raw_snp_position %>% select(SNP_ID, Chromosome, Position)
```
### 4.separate groups maize and teosinte
```{r}
genotypes_maize <- genotypes %>% 
  filter(Group %in% c("ZMMIL", "ZMMLR", "ZMMMR")) 

genotypes_teosinte <- genotypes %>% 
  filter(Group %in% c("ZMPBA", "ZMPIL", "ZMPJA"))
```
### 5. transpose data set and adjust rows and headers
```{r}
genotypes_maize_t <- genotypes_maize %>%
  t() %>%
  as.data.frame() %>%
  slice(-2) %>%
  mutate(Sample_ID = rownames(.)) %>%
  relocate(Sample_ID) %>%
  { colnames(.) <- .[1, ]; . } %>%
  slice(-1) %>%
  mutate(across(everything(), as.character)) %>%
  mutate(across(-1, as.factor))

rownames(genotypes_maize_t) <- NULL

genotypes_teosinte_t <- genotypes_teosinte %>%
  t() %>%
  as.data.frame() %>%
  slice(-2) %>%
  mutate(Sample_ID = rownames(.)) %>%
  relocate(Sample_ID) %>%
  { colnames(.) <- .[1, ]; . } %>%
  slice(-1) %>%
  mutate(across(everything(), as.character)) %>%
  mutate(across(-1, as.factor))

rownames(genotypes_teosinte_t) <- NULL

```
### 6.joint snp and genotypes
```{r}
joint_snp_maize <- snp_position %>%
  bind_cols(genotypes_maize_t[, -1])

joint_snp_teosinte <- snp_position %>% 
  bind_cols(genotypes_teosinte_t[, -1])
```
### 7. Generate 40 files
```{r}
# maize increase file generate
maize_chromosome_increase <- function(chr) {
  joint_snp_maize %>%
    filter(Chromosome == chr) %>%
    mutate(Numeric_Position = suppressWarnings(as.numeric(Position))) %>%  
    arrange(is.na(Numeric_Position), Numeric_Position) %>%  
    select(-Numeric_Position) %>%
    write_tsv(paste0("maize_data/maize_chrom", chr, "_increase.txt"))
}

lapply(1:10, maize_chromosome_increase)

# maize decrease file generate
maize_chromosome_decrease <- function(chr) {
  joint_snp_maize %>%
    filter(Chromosome == chr) %>%
    mutate(Numeric_Position = suppressWarnings(as.numeric(Position))) %>%  
    arrange(is.na(Numeric_Position), desc(Numeric_Position)) %>%  
    select(-Numeric_Position) %>%
    mutate(across(4:ncol(.), ~ ifelse(. == "?/?", "-/-", .))) %>% 
    mutate(across(4:ncol(.), as.character)) %>%
    write_tsv(paste0("maize_data/maize_chrom", chr, "_decrease.txt"))
}

lapply(1:10, maize_chromosome_decrease)


# teosinte increase file generate
teosinte_chromosome_increase <- function(chr) {
  joint_snp_teosinte %>%
    filter(Chromosome == chr) %>%
    mutate(Numeric_Position = suppressWarnings(as.numeric(Position))) %>%
    arrange(is.na(Numeric_Position), Numeric_Position) %>%
    select(-Numeric_Position) %>%
    write_tsv(paste0("teosinte_data/teosinte_chrom", chr, "_increase.txt"))
}

lapply(1:10, teosinte_chromosome_increase)

# teosinte decrease file generate
teosinte_chromosome_decrease <- function(chr) {
  joint_snp_teosinte %>%
    filter(Chromosome == chr) %>%
    mutate(Numeric_Position = suppressWarnings(as.numeric(Position))) %>%  
    arrange(is.na(Numeric_Position), desc(Numeric_Position)) %>%  
    select(-Numeric_Position) %>%  
    mutate(across(4:ncol(.), ~ ifelse(. == "?/?", "-/-", .))) %>%  
    mutate(across(4:ncol(.), as.character)) %>%
    write_tsv(paste0("teosinte_data/teosinte_chrom", chr, "_decrease.txt"))
}

lapply(1:10, teosinte_chromosome_decrease)

```
### 8.generate long file 
```{r}
genotypes_long <- raw_genotypes %>%
  pivot_longer(cols = -c(Sample_ID, JG_OTU, Group), names_to = "SNP_ID", values_to = "Genotype")

joint_snpALL_long <- genotypes_long %>%
  left_join(snp_position, by = "SNP_ID") %>% 
  filter(Group %in% c("ZMMIL", "ZMMLR", "ZMMMR","ZMPBA", "ZMPIL", "ZMPJA"))
```



